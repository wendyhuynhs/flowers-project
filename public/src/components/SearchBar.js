import React, { useState } from "react";
import { connect } from "react-redux";
import { Hint } from "react-autocomplete-hint";
import styles from "../styles/searchBar.scss";

const SearchBar = ({ searchPost, listOfAutofillOptions }) => {
  const [userInput, setUserInput] = useState("");

  const onInputChange = (e) => {
    e.persist();
    setUserInput(e.target.value);
  };

  const onFormSubmit = (e) => {
    e.preventDefault();
    searchPost(userInput);
  };

  return (
    <div className={styles.container}>
      <form onSubmit={onFormSubmit}>
        <Hint options={listOfAutofillOptions} allowTabFill="true">
          <input
            type="text"
            name="title"
            value={userInput}
            placeholder="Search title"
            onChange={onInputChange}
          />
        </Hint>
        <button>Submit</button>
      </form>
    </div>
  );
};
const mapStateToProps = ({ posts }) => {
  return { posts, listOfAutofillOptions: posts.map((post) => post.title) };
};

const mapDispatchToProps = (dispatch) => {
  return {
    searchPost: (input) => dispatch({ type: "SEARCH_TITLE", payload: input }),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchBar);
