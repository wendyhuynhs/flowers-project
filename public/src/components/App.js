import React from "react";
import PostList from "./PostList.js";
import SearchBar from "./SearchBar.js";

const App = () => {
  return (
    <div>
      <SearchBar />
      <PostList />
    </div>
  );
};

export default App;
