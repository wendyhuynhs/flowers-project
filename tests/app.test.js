import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import App from "../public/src/components/App";
import SearchBar from "../public/src/components/SearchBar";

const mockStore = configureMockStore([]);

describe("Rendering Components", () => {
  let store;
  let component;
  it("renders App component correctly", () => {
    const app = shallow(<App />);
    expect(app).toMatchSnapshot();
  });

  it("renders SearchBar component correctly", () => {
    const store = mockStore({});
    const app = shallow(
      <Provider store={store}>
        <SearchBar />
      </Provider>
    );
    expect(app).toMatchSnapshot();
  });
});
