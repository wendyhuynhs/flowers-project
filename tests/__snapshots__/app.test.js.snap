// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`App renders correctly 1`] = `ShallowWrapper {}`;

exports[`Rendering Components renders App component correctly 1`] = `ShallowWrapper {}`;

exports[`Rendering Components renders PostList component correctly 1`] = `ShallowWrapper {}`;

exports[`Rendering Components renders SearchBar component correctly 1`] = `ShallowWrapper {}`;
